sqlglot/transforms.py

```diff
@@ -133,6 +133,48 @@ def remove_precision_parameterized_types(expression: exp.Expression) -> exp.Expr
     )
 
 
+def decode_to_case(expression: exp.Expression) -> exp.Expression:
+    """
+    Transforms the Snowflake/Oracle/Redshift DECODE function into a CASE expression. Note that NULL
+    needs special treatment in CASE expressions, since we need to explicitly check for them using
+    the `IS NULL` predicate, instead of relying on pattern matching.
+
+    - https://stackoverflow.com/a/3209948/17518317
+    - https://docs.snowflake.com/en/sql-reference/functions/decode
+    """
+    if isinstance(expression, exp.Matches):
+        from sqlglot.optimizer.simplify import simplify
+
+        select = expression.this
```

Comment:
The name `select` here was picked to be consistent with Snowflake's documentation:

<img width="311" alt="Screenshot 2023-04-01 at 10 20 08 PM" src="https://user-images.githubusercontent.com/46752250/229310154-c71b5670-a2b7-4702-88a4-83f0bc7e616b.png">